src/native/libs/System.Native/pal_networking.c

```diff
@@ -1406,8 +1407,17 @@ int32_t SystemNative_ReceiveSocketError(intptr_t socket, MessageHeader* messageH
     messageHeader->ControlBuffer = (void*)buffer;
 
     struct msghdr header;
+    struct icmphdr icmph;
+    struct iovec iov;
     ConvertMessageHeaderToMsghdr(&header, messageHeader, fd);
 
+    if (header.msg_iovlen == 0 || !header.msg_iov)
```

Comment:
On RISC-V recvmsg fails to get non-empty message from error queue when msg_iov* fields are zeroed. On x64 it works fine. Since man pages are silent about what happen when iov data is empty (in our case recvmsg returns 0 so apparently it's not error) I assume that behavior is implementation defined. After quick look it seems that iov is processed in skb_copy_datagram_msg function https://elixir.bootlin.com/linux/latest/source/net/ipv4/ip_sockglue.c#L545 and then number of copied bytes is what recvmsg eventually returns. Anyway, we follow ping here and fill iov data with protocol related structures before getting message from error queue.